### PR TITLE
add login silent implementation

### LINKED ISSUE #1: Login silently implementation
When integrated into the app, even after a successful login, this module will ask for credentials again due to the lack of silent login function.

### PR DESCRIPTION
**Pull request by unpluggedk**
fixed #1

### CODE REVIEW

File: MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m [hunk 1]
```diff
@@ -188,9 +188,17 @@ - (void)loginWithViewController:(UIViewController*)viewController
...
-//    // Placeholder: Load an existing account from the account store
-//}
+- (BOOL)loginSilent {
+    // Placeholder: Load an existing account from the account store
+    int accountCount = [[NXOAuth2AccountStore sharedStore] accounts].count;
+    
+    if (accountCount == 0) {
+        return NO;
+    } else {
+        self.userAccount = [[NXOAuth2AccountStore sharedStore] accounts][accountCount - 1];
```
Comment (kevklam) [inline MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m]: Is NXOauth2 documented to rearrange its account list in MRU fashion? If not it might be prudent to track that in shared preferences or something.

File: MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m [hunk 2]
```diff
@@ -188,9 +188,17 @@ - (void)loginWithViewController:(UIViewController*)viewController
     }];
 }
 
-//- (BOOL)loginSilent {
-//    // Placeholder: Load an existing account from the account store
-//}
+- (BOOL)loginSilent {
+    // Placeholder: Load an existing account from the account store
```
Comment (kevklam) [inline MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m]: stray comment

File: MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.h [hunk 3]
```diff
@@ -66,7 +66,7 @@
  Silent login is not implemented in this example auth provider, but if implemented would
```
Comment (kevklam) [inline MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.h]: stray comment

**Comment by kevklam** [comment]
🕥

File: MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m (hunk 1 above)
Comment (unpluggedk) [inline MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m]: Nope it does not. 
In fact, I believe it is using a dictionary with account identifier. We can be more "accurate" by storing "last logged in" identifier and using that value to really fetch the last successful login. 

I just thought as a sample, with an assumption of a single user account login, this would suffice. 

Let me know.

**Comment by unpluggedk** [comment]
And comments will be deleted

File: MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m (hunk 1 above)
Comment (kevklam) [inline MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m]: Yeah, it's a sample but more in the sense of incomplete functionality - anything we do include should be as correct and unsurprising to the developer as possible.

Comment (unpluggedk) [inline MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m]: Aight, I could throw in a user preference to hold the identifier and get the very last one.
Will do another pull request shortly, let me know what you think about that approach.

Comment (kevklam) [inline MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m]: Sounds good to me.

File: MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m [hunk 4]
```diff
@@ -26,6 +26,8 @@
 #import "MSAuthenticationViewController.h"
 #import "MSLogger.h"
 
+static NSString *const lastIdentifierKey = @"lastIdentifierKey";
```
Comment (kevklam) [inline MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m]: Unique key like "com.graph.nxoauth2.lastIdentifierKey" to avoid collisions with consuming app

**Comment by kevklam** [comment]
:shipit:

**Comment by unpluggedk** [comment]
Done, agreed.
